<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <title>spatialGE.STgradient.Launcher</title>
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

        <!-- Import Bootstrap -->
        <link rel="stylesheet" href="visualizer/bootstrap/css/bootstrap.min.css">
        <script src="/gp/visualizer/v1/jquery/jquery-3.7.1.min.js"></script>
        <script src="/gp/visualizer/v1/bootstrap/js/popper.min.js"></script>
        <script src="/gp/visualizer/v1/bootstrap/js/bootstrap.min.js"></script>

        <!-- Load Vue -->
        <script src="/gp/visualizer/v1/vue/vue.global.min.js"></script>

        <!-- Import Font Awesome -->
        <link href="/gp/visualizer/v1/fontawesome/css/fontawesome.css" rel="stylesheet" />
        <link href="/gp/visualizer/v1/fontawesome/css/solid.css" rel="stylesheet" />
        
        <style>
            #thumbs {
                overflow-x: auto;
                overflow-y: hidden;
                white-space: nowrap;
            }
            
            #thumbs img {
                width: 100px;
                height: 100px;
                margin-right: 10px;
                margin-bottom: 10px;
                cursor: pointer;
            }
        </style>
    </head>
    <body id="app" >
        <form id="form-collapse" class="container mt-3" @submit.prevent="handle_submit">
            <div class="row">
                <h3>Spatial gradients with STgradient</h3>
                <p>STgradient tests for genes for which there is evidence of expression spatial gradients with respect to a
                    “reference” tissue niche/domain (e.g., higher expression closer to reference tissue niche, lower
                    expression as farther from reference tissue niche).</p>
                <p>The method calculates distances from each spot/cell
                    to the reference tissue niche/domain (e.g., a cluster defined via spatialGE.STclust) and correlates
                    those distances with gene expression values from top variable genes (defined
                    by standard deviation across ROIs/spots/cells). The distances to reference niche can be summarized using
                    the average or the minimum value. Generally, the minimum distances might be better to capture gradients
                    at short ranges, while average distances capture whole-tissue gradients. Use robust regression to
                    reduce (albeit not eliminate) the effect of zero inflation in spatial transcriptomics data. The most
                    variable genes to be tested are identified before removal of outliers.</p>
            </div>
            <div class="row">
                <div id="error" class="alert alert-danger col-md-12 d-none"></div>
            </div>

            <div class="row">
                <h4>Select samples to include</h4>
                <table id="samples" class="table table-striped">
                    <thead><tr>
                        <th><input id="check-all" type="checkbox" @click="toggle_samples"></th>
                        <th>Sample Name</th>
                        <th>Total spots/cells</th>
                        <th>Total genes</th>
                        <th>Min. counts</th>
                        <th>Avg. counts</th>
                        <th>Max. counts</th>
                        <th>Min. genes</th>
                        <th>Avg. genes</th>
                        <th>Max. genes</th>
                    </tr></thead>
                    <tbody>
                        <tr id="sample-loading">
                            <td colspan="10"><h3 class="text-center"><i class="fa-solid fa-spinner fa-spin"></i></h3></td>
                        </tr>
                        <tr v-for="(sample, index) in samples" :key="index">
                            <td><input type="checkbox" class="form-control form-check" :name="`${sample.sample_name}`" :id="`sample-${index}`" :value="sample.value" v-model="sample.selected"></td>
                            <td>{{ sample.sample_name }}</td>
                            <td>{{ sample.spotscells }}</td>
                            <td>{{ sample.genes }}</td>
                            <td>{{ sample.min_counts_per_spotcell }}</td>
                            <td>{{ sample.mean_counts_per_spotcell }}</td>
                            <td>{{ sample.max_counts_per_spotcell }}</td>
                            <td>{{ sample.min_genes_per_spotcell }}</td>
                            <td>{{ sample.mean_genes_per_spotcell }}</td>
                            <td>{{ sample.max_genes_per_spotcell }}</td>
                        </tr>
                    </tbody>
                </table>
            </div>

            <div class="row mt-3">
                <div class="col-md-4">
                    <label class="form-label" for="numVariableGenes"><strong>Number of most variable genes to use</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="Maximum number of genes per sample to test for spatial gradients. The genes to be tested are selected based on standard deviation."></i></label>
                    <input id="numVariableGenes" class="form-control" type="number" step="1" min="0" v-model.number="form.numVariableGenes" />
                </div>

                <div class="col-md-4">
                    <label for="numVariableGenesSlider" class="form-label">&nbsp</label>
                    <input type="range" class="form-control form-range" id="numVariableGenesSlider"step="1" min="0" max="3000" v-model.number="form.numVariableGenes" >
                </div>
            </div>

            <div class="row mt-3">
                <div class="col-md-4">
                    <label class="form-label"><strong>Annotation to test</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="The tissue niches on which to run STgradient. This parameter should be set to the name of a column in @spatial_meta containing the tissue domain assignments for each spot or cell. These are generated by STclust."></i></label>
                    <select id="annotation" class="form-control form-select" v-model="form.annotationToTest">
                        <option v-for="annotation in annotations" :key="annotation.value" :value="annotation.value">
                            {{ annotation.label }}
                        </option>
                    </select>
                </div>
                <div class="col-md-4">
                    <label class="form-label"><strong>Reference cluster</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="Reference cluster from which spatial gradients are to be tested. The reference cluster could be, for example, a tumor region detected by STclust. One of the tissue domains in the niche specified in annotation to test. Corresponds to the 'reference' cluster or domain. Spearman's correlations will be calculated using spots assigned to domains other than this reference domain (or domains specified in exclude clusters)."></i></label>
                    <select id="reference" class="form-control form-select" required v-model.number="form.referenceCluster">
                        <option v-for="n in referenceClusterRange" :key="n.cluster" :value="n.cluster">{{ n.cluster }}</option>
                    </select>
                </div>
                <div class="col-md-4">
                    <label class="form-label"><strong>Cluster to exclude (optional)</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="Comma-separated list of clusters to exclude. Exclude regions/niches from the analysis. By specifying one or more niches, the distances of those spots/cells will not be calculated and will be excluded from the analyses. This option could be useful when removing tissue niches that show necrosis and could add noise to the Spearman's coefficients."></i></label>
                    <select id="exclude" class="form-control form-select" v-model="form.excludeClusters">
                        <option v-for="n in excludeClusterRange" :key="n.cluster" :value="n.cluster">{{ n.cluster }}</option>
                    </select>
                </div>
            </div>

            <div class="row mt-3">
                <div id="thumbs" class="col-md-12">
                </div>
            </div>

            <div class="row mt-3">
                <div class="col-md-4">
                    <!-- Checkbox for robust.regression -->
                    <div class="form-check mb-2 mr-sm-2">
                        <input type="checkbox" class="form-check-input" v-model="form.robustRegression" />
                        <label class="form-check-label"><strong>Robust regression</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="In robust regression, outliers are given less weight towards the calculation of the regression coefficient."></i></label>
                    </div>
                </div>
                <div class="col-md-4">
                    <!-- Checkbox for ignore.outliers -->
                    <div class="form-check mb-2 mr-sm-2">
                        <input type="checkbox" class="form-check-input" v-model="form.ignoreOutliers" :disabled="form.robustRegression" />
                        <label class="form-check-label"><strong>Ignore outliers</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="Whether to ignore outliers. This option is ignored if robust regression is selected. If outliers are ignored, traditional linear regression is carried out after removing spots/cells defined as outliers by the interquartile range method. If set to false, all spots are considered in the analysis."></i></label>
                    </div>
                </div>
                <div class="col-md-4">
                </div>
            </div>

            <div class="row mt-3">
                <div class="col-md-4">
                    <label class="form-label" for="correlationLimit"><strong>Restrict correlation to this limit</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="A distance value to restrict the correlation analysis. Some tissues can be very heterogeneous in composition. As a result it could be reasonable to test for spatial gradients within a restricted area of the tissue."></i></label>
                    <input id="correlationLimit" class="form-control" type="number" step="0.01" v-model.number="form.correlationLimit" />
                </div>
                <div class="col-md-4">
                    <label class="form-label" for="minNeighbors"><strong>Minimum number of neighbors</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="The minimum number of immediate neighbors a reference spot/cell must have to be included in the analysis. This parameter intends to reduce the effect of isolated spots/cells in the calculation of the correlation coefficients. Unsupervised clustering algorithms (such as STclust) can assign niches to isolated spots. Reference spots with less neighbors than specified, will be omitted from the analysis, and hence distances from that spot will not be calculated."></i></label>
                    <input id="minNeighbors" class="form-control" type="number" min="1" step="1" required v-model.number="form.minNeighbors" />
                </div>
                <div class="col-md-4">
                    <label class="form-label" for="distanceSummary"><strong>Distance summary metric</strong> <i class="fa-solid fa-circle-info text-muted" data-toggle="tooltip" title="Distance summary metric: 'average' or 'minimum'"></i></label>
                    <select id="distanceSummary" class="form-control form-select" v-model="form.distanceSummary">
                        <option v-for="(key, value) in distanceSummaryOptions" :key="key" :value="key">
                            {{ value }}
                        </option>
                    </select>
                </div>
            </div>

            <div class="row mt-3 mb-5">
                <div class="col-md-4">
                    <button class="btn btn-lg btn-primary" type="submit">Run STgradient</button>
                </div>
                <div class="col-md-4 mt-2">
                </div>
                <div class="col-md-4"></div>
            </div>
        </form>
        <div id="results-collapse" class="container mt-3 d-none">
            <nav class="navbar navbar-dark bg-dark row rounded mt-3">
                <div class="col-md-4 mt-2">
                    <h3 id="job-status" class="navbar-brand font-weight-bold"></h3>
                </div>
                <div class="col-md-8">
                    <p class="lead text-white mt-3 text-right">Submitted jobs may take several minutes to an hour to complete.</p>
                </div>
            </nav>
            <div class="row mt-3 mb-3">
                <div id="job-results" class="col-md-12">
                </div>
            </div>
        </div>

        <script type="module" src="js/app.js"></script>
        <script>
            $(document).ready(() => {
                $('[data-toggle="tooltip"]').tooltip({placement: 'top'});
                $('#form-collapse').collapse();
            });
        </script>
    </body>
</html>